packages/lib/holidays/GoogleHolidayService.ts

```diff
@@ -0,0 +1,216 @@
+import dayjs from "@calcom/dayjs";
+import prisma from "@calcom/prisma";
+
+import { GOOGLE_HOLIDAY_CALENDARS, getHolidayCacheDays } from "./constants";
+
+interface GoogleCalendarEvent {
+  id: string;
+  summary: string;
+  description?: string;
+  start: {
+    date?: string;
+    dateTime?: string;
+  };
+  end: {
+    date?: string;
+    dateTime?: string;
+  };
+}
+
+interface GoogleCalendarEventsResponse {
+  items?: GoogleCalendarEvent[];
+  error?: {
+    code: number;
+    message: string;
+  };
+}
+
+export interface CachedHoliday {
+  id: string;
+  countryCode: string;
+  eventId: string;
+  name: string;
+  date: Date;
+  year: number;
+}
+
+class GoogleHolidayServiceClass {
+  private apiKey: string | null = null;
+
+  private getApiKey(): string {
+    if (this.apiKey) return this.apiKey;
+
+    const key = process.env.GOOGLE_CALENDAR_API_KEY;
+    if (!key) {
+      throw new Error("GOOGLE_CALENDAR_API_KEY environment variable is not set");
+    }
+    this.apiKey = key;
+    return key;
+  }
+
+  async fetchHolidaysFromGoogle(countryCode: string, year: number): Promise<CachedHoliday[]> {
+    const calendarConfig = GOOGLE_HOLIDAY_CALENDARS[countryCode];
+    if (!calendarConfig) {
+      return [];
+    }
+
+    const apiKey = this.getApiKey();
+    const calendarId = encodeURIComponent(calendarConfig.calendarId);
+
+    const timeMin = dayjs(`${year}-01-01`).startOf("day").toISOString();
+    const timeMax = dayjs(`${year}-12-31`).endOf("day").toISOString();
+
+    const url = `https://www.googleapis.com/calendar/v3/calendars/${calendarId}/events?key=${apiKey}&timeMin=${timeMin}&timeMax=${timeMax}&singleEvents=true&orderBy=startTime`;
+
+    const response = await fetch(url);
+    const data: GoogleCalendarEventsResponse = await response.json();
+
+    if (data.error) {
+      console.error(`Google Calendar API error for ${countryCode}:`, data.error);
+      throw new Error(`Google Calendar API error: ${data.error.message}`);
+    }
+
+    if (!data.items) {
+      return [];
+    }
+
+    return data.items
+      .filter((event) => {
+        const desc = event.description?.toLowerCase() || "";
+        return desc.includes("public holiday");
```

Comment:
I'm asking with no knowledge or clue at all, but is it guaranteed that they include "public holiday" text in the description? (for non-English speaking countries?)